controller/api/destination/server.go

```diff
@@ -268,11 +224,38 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 			return nil
 		}
 	} else {
-		service, _, err = parseK8sServiceName(host, s.clusterDomain)
+		var podName string
+		service, podName, err = parseK8sServiceName(host, s.clusterDomain)
 		if err != nil {
 			log.Debugf("Invalid service %s", path)
 			return status.Errorf(codes.InvalidArgument, "invalid service: %s", err)
 		}
+
+		// If the pod name (instance ID) is not empty, it means we parsed a DNS
+		// name. When we fetch the profile using a pod's DNS name, we want to
+		// return an endpoint in the profile response.
+		if podName != "" {
+			pod, err := s.k8sAPI.Pod().Lister().Pods(service.Namespace).Get(podName)
```

Comment:
I believe that the hostname of an endpoint can be something other than the pod name (even though pod name is the default).  I think what we actually need to do here is fetch the Endpoints object for the service, iterate through the endpoints and find the one who's hostname matches the instanceID we parsed from the fqdn.  And then fetch the pod in that endpoint's targetRef (if it has one).

Reply:
Makes sense, this completely eluded me. For anyone interested, a [pod spec includes a hostname field](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields). While generally, the hostname will correspond to the value of the pod's `.metadata.name` field, it can be overwritten.

Will make the changes asap! :) 

https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields

Reply:
On second thought @adleong, since pods may not map to Endpoints it would also feel more correct to build the endpoint directly from the Endpoints object (rather than fetch the pod and then build the weighted address based on that). Although we may not be able to re-use as much code, would it be a better idea to build the weighted address with the IP from the Endpoints object?